Remote notifications

faysal-ali opened the pull request.

faysal-ali committed:
Push Notification commits squashed

faysal-ali committed:
Push Notification commits squashed

faysal-ali committed:
Push Notification commits squashed

amulji committed:
Removed fundraising cell and challenge progress cell from dashboard

amulji committed:
removed commented code

amulji committed:
Tempory Fix for commitment update from settings page. Showing alert with message "No event available"

amulji committed:
Journey list steps updated to miles

faysal-ali committed:
Merge branch 'remote-notifications' of https://github.com/faysal-ali/WCF-iOS into remote-notifications

faysal-ali committed:
Notification API parsing

faysal-ali committed:
Notification API parsing

faysal-ali committed:
Renamed old Notification struct to NotificationV1

faysal-ali committed:
Renamed old Notification struct to NotificationV1

faysal-ali committed:
Added formattedMessageDate

faysal-ali committed:
Added formattedMessageDate

faysal-ali committed:
Added notificationCell instead of reusing InfoCell as UI is different

faysal-ali committed:
Added notificationCell instead of reusing InfoCell as UI is different

faysal-ali committed:
Fixed warnings and using identifier instead of notificationId

faysal-ali committed:
Fixed warnings and using identifier instead of notificationId

faysal-ali committed:
Fixed crash

faysal-ali committed:
Fixed crash

amulji committed:
Create Team complete

amulji committed:
Added teamName is invite memebers message + removed comments

amulji committed:
Azure image upload complete + badges daily streak assets added

amulji committed:
pod install AZSClient

compnerd commented on Steps4Impact/App/AppController.swift:
```diff
@@ -245,4 +253,31 @@ class AppController {
+
+  func registerForRemoteNotifications() {
+    UIApplication.shared.registerForRemoteNotifications()
+    UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .sound, .badge]) { granted, _ in
```
I think that using a `nil` completion is better.  The logs are not going to be available to us as telemetry and so they are just lost.

compnerd commented on Steps4Impact/App/AppDelegate.swift:
```diff
@@ -47,4 +47,19 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
+
+  func application(_ app: UIApplication,
+                   didFailToRegisterForRemoteNotificationsWithError error: Error) {
+    print("Failed to register remote notifications", error)
```
Debugging left overs?

compnerd commented on Steps4Impact/App/Navigation.swift:
```diff
@@ -60,10 +60,19 @@ class Navigation: UITabBarController {
     self.viewControllers = [dashboard, challenge, leaderboard, notifications]
+
+    _ = NotificationCenter.default.addObserver(forName: .receivedNotification,
+                                           object: nil,
```
Indentation seems off

compnerd commented on Steps4Impact/DashboardV2/DashboardViewController.swift:
```diff
@@ -62,10 +64,40 @@ class DashboardViewController: TableViewController {
+          message: Strings.NotificationsPermission.message, preferredStyle: .alert)
+        controller.addAction(UIAlertAction(
+          title: Strings.NotificationsPermission.proceed, style: .default,
+          handler: { (_) in
```
The tuple in the closure is unnecessary, just blackhole the argument.  Extracting the action with the trailing closure is nicer.

compnerd commented on Steps4Impact/Models/Notification.swift:
```diff
@@ -0,0 +1,67 @@
+//
+//  Created by Faisal Ali on 1/21/20.
+//  Copyright © 2020 AKDN. All rights reserved.
+//
```
Please use the standard header

compnerd commented on Steps4Impact/Notifications/Notifications.swift:
```diff
@@ -66,14 +66,14 @@ class NotificationView: UITableViewCell {
 }
 
-struct Notification {
+struct NotificationV1 {
```
The entire reason to have the `V1` suffix is to leave the old type in place.  If you are completely replacing it, just keep the old name.

compnerd commented on Steps4Impact/NotificationsV2/NotificationsViewController.swift:
```diff
@@ -30,22 +30,222 @@
+        appropriateFor: nil, create: false)
+      return documentURL.appendingPathComponent(filename)
+    } catch {
+      print(error)
```
debugging?

compnerd commented on Steps4Impact/Support/UIColor.swift:
```diff
@@ -64,6 +64,18 @@ extension UIColor {
+  static var unread: UIColor {
+    get {
+      return UIColor.init(displayP3Red: 234.0/255.0, green: 244.0/255.0, blue: 239.0/255.0, alpha: 1.0)
+    }
```
Can you please create static constants in `Style` for the colors?

samisuteria committed:
bump build number

samisuteria committed:
bump target version to 12

samisuteria committed:
add helper

samisuteria committed:
update connection settings

samisuteria committed:
use silver

samisuteria committed:
remove deprecated api for ios 12

samisuteria committed:
update health kit data provider to use units and new auth api

samisuteria committed:
keep a reference to provider (losing ref in closure)

samisuteria committed:
update ui

samisuteria committed:
bump build number

samisuteria committed:
update localization test

aalimmulji committed:
Amulji/new updates (#362)

* Added testRun toggle
Added testRun milestone data
Fixed total milestones number
Small fix in Create Team

* Updated Current build to 271

* TeamImage download on view team page complete

* Team edit image complete

* Updated final medal badge logic to be dynamic with the event duration

* Added badges assets + minor fixes

* added missing localize string

aalimmulji committed:
Amulji/journey detail view fixes (#363)

* page-control complete

* page control complete

* Temp fixes

* Added testRun toggle
Added testRun milestone data
Fixed total milestones number
Small fix in Create Team

(cherry picked from commit b209869ccd62eb07834525b2c4414d3264c2460f)

* removed comments

* uncommented appconfig settings for debug

* commented fundraising cell on challenge page

* bump build number

aalimmulji committed:
Some more changes (#364)

* Remove badges Image View border

* Invite new members allow only to team lead

* Added activityView to tableViewVC and BadgesCollectionVC + moved journeyDetailVC to journey folder

* minor fixes for badgesVC and Commitment Card on ChallegeVC

* added notification observer for team changed on leaderboardVC

* Hidding journeyDetailVC temporary until fixed

aalimmulji committed:
Team breakdown complete (#367)

samisuteria committed:
Activity Card Updates (#366)

* add chevron assets

* add extensions

* lots of views

* update protocols for pedometer

* don’t send records till fitbit is fixed

aalimmulji committed:
Journey detail fix (#365)

* build++

* Team selected show checkmark on select

* Added helper func to convert html to string + create hyperlink

* push JourneyDetailVC

* turned off dark mode

* New share button for journey detail

* Bump builder number

amulji committed:
minor fixes

samisuteria committed:
podfile fixes

samisuteria committed:
Fitbit Provider Fixes (#370)

* add dateformatter extension

* update model to store date instead of string

* update fitbit provider with new protocol

aalimmulji committed:
Challenge progress (#371)

* minor fixes

* Challenge progress complete

* Team Settings view team fix

* Team Breakdown fixes

* build number ++

faysal-ali committed:
Update swift-lint. Fixed warnings

faysal-ali committed:
Update swift-lint. Fixed warnings

faysal-ali committed:
Merge branch 'remote-notifications' of https://github.com/faysal-ali/WCF-iOS into remote-notifications

# Conflicts:
#	Steps4Impact/Journey/Cells/CurrentMilestoneCell.swift
#	Steps4Impact/Support/UITableView.swift
